core/eolearn/core/eoexecution.py

```diff
@@ -129,14 +145,15 @@ def run(self, workers: int = 1, multiprocess: bool = True) -> List[WorkflowResul
         self.start_time = dt.datetime.now()
         self.report_folder = self._get_report_folder()
         if self.save_logs:
-            os.makedirs(self.report_folder, exist_ok=True)
+            self.filesystem.makedirs(self.report_folder, recreate=True)
 
-        log_paths = self.get_log_paths() if self.save_logs else [None] * len(self.execution_args)
+        log_paths = self.get_log_paths(full_path=True) if self.save_logs else [None] * len(self.execution_kwargs)
 
         filter_logs_by_thread = not multiprocess and workers > 1
-        processing_args = [(self.workflow, init_args, log_path, filter_logs_by_thread, self.logs_filter)
-                           for init_args, log_path in zip(self.execution_args, log_paths)]
         processing_type = self._get_processing_type(workers, multiprocess)
+        processing_args = [
```

Comment:
This is really becoming slightly unwieldy. I would again suggest having a dataclass, either for all parameters or just for the ones pertaining to logging

As a further point, mypy is detecting an error, because some of the arguments are allegedly `Optional` but the function signatures does not have them listed as optional

Reply:
Done - changed to a dataclass.